docs/source/traitsui_user_manual/testing.rst

```diff
@@ -333,9 +333,27 @@ it with any testing framework (e.g. unittest, pytest).
 
 .. rubric:: Is UITester compatible with PyFace ModalDialogTester?
 
-Yes. For example, you can use |UITester| to launch a modal dialog, and use
+Yes, with some care.
+
+For example, you can use |UITester| to launch a modal dialog, and use
 ModelDialogTester to close it.
 
+But if you are attempting to modify or inspect GUI states using |UITester|
+while the dialog is opened, you should set the ``process_events`` attribute to
+false for those operations. Otherwise the ModalDialogTester and UITester
+will enter a race condition that blocks forever.
+
+Example::
+
+    def when_opened(modal_dialog_tester):
+        ui_tester = UITester(process_events=False)
+        ui_tester.find_by_id(ui, "button").perform(MouseClick())
```

Comment:
Idk if it is necessary, but it might be confusing where this `ui` comes from seeing as we don't use `with tester.create_ui() as ui:` here. We could add a brief sentence explaining that you will need access to it via a different route (e.g. I found myself getting it from modal_dialog_tester.get_dialog_widget()._ui).

Reply:
I added a sentence under the example, hope that looks okay.